### PR TITLE
No std rstd

### PR DESCRIPTION
**Pull request by brenzi**
This adds `no_std` support. (without depending on `sgx_tstd`)
In order for this to work, I had to fork `substrate-primitives` and enable crypto primitives.
This also drops dependency on custom runtime crate.

TODO:
* `example_contract` is broken. blocker: #26

### CODE REVIEW

**Comment by brenzi** [comment]
I suggest to merge although example_contract is broken. If someone wants to use it, I've added a hint

**Comment by brenzi** [comment]
the recent merge commit broke no_std, need to fix this first

**Comment by brenzi** [comment]
no_std fix works

File: src/extrinsic/mod.rs [hunk 1]
```diff
@@ -22,8 +22,7 @@ pub mod balances;
 pub mod contract;
 pub mod xt_primitives;
 
-pub extern crate codec;
-pub extern crate log;
```
Comment (clangenb) [inline src/extrinsic/mod.rs]: Has this been done for a reason?

I made this for issue #16 . This re-exports the dependencies locally. Making them available in the macro via `crate::...`.

Then third parties using `substrate-api-client` do not need to import the `codec` and `log` dependencies

**Comment by clangenb** [comment]
apart from my comment i think this is ready to merge.

File: src/extrinsic/mod.rs (hunk 1 above)
Comment (brenzi) [reply src/extrinsic/mod.rs]: the reason for removing it is that it breaks `no_std`

Comment (clangenb) [reply src/extrinsic/mod.rs]: try reexporting conditionally, only if in `std` environment

**Pull request merged**